### DISCUSSION TITLE
vscode-f5 2024 roadmap ideas

### DISCUSSION THREAD

**Discussion by DumpySquare** [DumpySquare/vscode-f5-fast#234] [Polls]
Looking for feedback on where to focus vscode features/dev for 2024.

Also open to any other ideas not in the list.  Please provide feedback;

- XC extension
  - connect with and manage tenants/creds
  - utilize APIs to list/create/modify applications
  - help with application lifecycle
  - copy/migrate/move applications/policies/objects
- TMOS to XC conversion
  - help with converting TMOS configs to XC configs
- vscode-f5-flipper (Citrix NS)
  - continue work on AS3 conversion process
  - NS to XC conversion flow
- NEXT
  - connect with and manage configs/apps like existing TMOS
- FAST extension
  - specific to managing only FAST templates
  - Live authoring view to see changes in real time
  - FAST template authoring will become a critical piece of the transition to NEXT
- Policy Supervisor integration
  - get/push waf policies from different products
  - interact with policy supervisor engine for conversions
  - push updated policy to different products
- Swagger file browser execution
  - This would take a swagger file from NEXT and provide a browsable interface to see all the APIs available
  - Could then provide connecitivty to f5 products to execute those api calls

**Comment by f5-rahm** [comment]
Oooh...that should be a ranked list! XC for sure, but getting FAST into the hands of BIG-IP Next early adopters/tire-kickers would be great too.

**Comment by JuergenMang** [comment]
I have voted for FAST, but WAF supervisor integration would be also very great!
